### DISCUSSION TITLE
Router organization

### DISCUSSION THREAD

**Discussion by kellytk** [tokio-rs/axum#382] [Q&A] [answered]
If I'd like to move a router definition out of the file it's used (`into_make_service()`) in, what's the recommended form for that to take?

For example, if I created a function in another file to return it:

```
pub fn router() -> Router {
Router::new()
    .route(...)
    .route(...)
    ...
}
```

what should the function's return value be?  `Router` is apparently incorrect and/or insufficient.

**Comment by davidpdrsn** [comment, marked as answer]
https://docs.rs/axum/0.2.8/axum/struct.Router.html#method.boxed
